include/zephyr/modem/chat.h

```diff
@@ -376,6 +376,156 @@ void modem_chat_script_abort(struct modem_chat *chat);
+ * @param chat_match Modem chat match instance
+ * @param match Match to set
+ * @note The lifetime of match must match or exceed the lifetime of chat_match
+ * @warning Always call this API after match is modified
+ *
+ * @retval 0 if successful
+ * @retval -ENOMEM if strlen of match exceeds UINT8_MAX
+ */
+int modem_chat_match_set_match(struct modem_chat_match *chat_match, const char *match);
+
+/**
+ * @brief Set separators of modem chat match instance
+ * @param chat_match Modem chat match instance
+ * @param separators Separators to set
+ * @note The lifetime of separators must match or exceed the lifetime of chat_match
+ * @warning Always call this API after separators is modified
+ *
+ * @retval 0 if successful
+ * @retval -ENOMEM if strlen of separators exceeds UINT8_MAX
+ */
+int modem_chat_match_set_separators(struct modem_chat_match *chat_match, const char *separators);
+
+/**
+ * @brief Set modem chat match callback
+ * @param chat_match Modem chat match instance
+ * @param callback Callback to set
+ */
+void modem_chat_match_set_callback(struct modem_chat_match *chat_match,
+				   modem_chat_match_callback callback);
+
+/**
+ * @brief Set modem chat match partial flag
+ * @param chat_match Modem chat match instance
+ * @param partial Partial flag to set
+ */
+void modem_chat_match_set_partial(struct modem_chat_match *chat_match, bool partial);
+
+/**
+ * @brief Set modem chat match wildcards flag
+ * @param chat_match Modem chat match instance
+ * @param wildcards Wildcards flag to set
+ */
+void modem_chat_match_set_wildcards(struct modem_chat_match *chat_match, bool wildcards);
+
+/**
+ * @brief Initialize modem chat script chat
+ * @param script_chat Modem chat script chat instance
+ */
+void modem_chat_script_chat_init(struct modem_chat_script_chat *script_chat);
+
+/**
+ * @brief Set request of modem chat script chat instance
+ * @param script_chat Modem chat script chat instance
+ * @param request Request to set
+ * @note The lifetime of request must match or exceed the lifetime of script_chat
+ * @warning Always call this API after Request is modified
+ *
+ * @retval 0 if successful
+ * @retval -ENOMEM if strlen of request exceeds UINT16_MAX
+ */
+int modem_chat_script_chat_set_request(struct modem_chat_script_chat *script_chat,
+				       const char *request);
+
+/**
+ * @brief Set modem chat script chat matches
+ * @param script_chat Modem chat script chat instance
+ * @param response_matches Response match array to set
+ * @param response_matches_size Size of response match array
+ * @note The lifetime of response_matches must match or exceed the lifetime of script_chat
+ *
+ * @retval 0 if successful
+ * @retval -EINVAL if response_matches_size > 0 and response_matches == NULL
+ */
+int modem_chat_script_chat_set_matches(struct modem_chat_script_chat *script_chat,
+				       const struct modem_chat_match *response_matches,
+				       uint16_t response_matches_size);
+
+/**
+ * @brief Set modem chat script chat matches
```

Comment:
Ahem? :slightly_smiling_face: 
I am questioning a bit the value of these comments...

Reply:
Most of them are quite self explanatory from the name of the function, I add the brief for a bit more grammatically correct documentation, I agree it may be a bit overkill/redundant, but it

Reply:
oh, got distracted :D ... but it is harmless :)